.github/workflows/scripts/comments.mjs

```diff
@@ -1,14 +1,70 @@
+import { readFile } from 'node:fs/promises'
+
+/**
+ * Posts the content of a diff as a comment on a Github issue
+ *
+ * @param {GithubActionContext} githubActionContext
+ * @param {number} issueNumber
+ * @param {DiffComment} diffComment
+ */
+export async function commentDiff (
+  githubActionContext,
+  issueNumber,
+  { path, title, markerText }
+) {
+  // Read diff from previous step
+  const diffText = await readFile(path, 'utf8')
+
+  // Add or update comment on PR
+  try {
+    await comment(
+      githubActionContext,
+      issueNumber,
```

Comment:
We've used a nice **DiffComment** object above `{ path, title, markerText }`

Should we add an optional `[bodyText]` to the type and merge it in? So we're being consistent

```js
await comment(githubActionContext, { ...diffComment, bodyText })
```

Reply:
I don't think we should merge the two things, `comment` doesn't really has a reason to accept a `path` (which we would pass by spreading `...diffComment`) given its role is just to post a comment on Github. TypeScript would likely complain about that as well, wouldn't it?

Reply:
Excuse the TypeScript code but with two comment flavours:

```ts
type IssueComment = {
  issue: number
  title: string
  bodyText: string
}

type DiffComment = {
  path: string
  markerText: string
  bodyText?: string
} & IssueComment
```

That gives me a way of writing comments that already have `bodyText` populated

```mjs
/**
 * 
 * @param {GithubActionContext} context 
 * @param {IssueComment | DiffComment} comment
 */
async function comment(context, comment) {
  const { github } = context

  // Detect diff comments with paths
  if (!comment.bodyText && 'path' in comment) {
    comment.bodyText = 'Load from disk example'
  }

  github.rest.issues.createComment({
    /* Call API example */
  })
}
```